docs/source/user-guide/vision/VisionData.rst

```diff
@@ -0,0 +1,184 @@
+.. _vision_data_class:
+
+========================
+The Vision Data Class
+========================
+The :class:`VisionData <deepchecks.vision>` data class is the base deepchecks class for
+representing the data for a vision task. It is essentially a wrapper around a batch loader of images, labels,
+and predictions in :doc:`Deepchecks' format </user-guide/vision/supported_tasks_and_formats>`
+which is used as the input for the different :doc:`vision checks </checks_gallery/vision>`.
+In addition it stores cache regarding the data, labels and predictions it encountered during the check run.
+
+
+Class Properties
+================
+
+The common properties are:
+
+- **batch_loader** - The batch loader can be either a pytorch DataLoader, a tensorflow Dataset or any custom
+  generator. It is important to note that the data loaded by the
```

Comment:
why use the word generator and not iterator?

Reply:
the return type of the function will be - t.Generator